dataflow/src/main/java/org/checkerframework/dataflow/analysis/AnalysisResult.java

```diff
@@ -195,6 +201,52 @@ private static void mergeTreeLookup(
         return merged;
     }
 
+    /**
+     * Returns true if the given tree is dead code (will never be executed at run time).
+     *
+     * <p>This method is conservative. If it returns true, the tree is dead code. If it returns
+     * false, the tree may or may not be dead code.
+     *
+     * @param tree a tree
+     * @return true if the tree is dead code
+     */
+    public boolean isDeadCode(Tree tree) {
+        if (!(tree instanceof ExpressionTree)) {
```

Comment:
You could use TreeUtils#isExpressionTree. (Though the implementation of that method is `tree instanceof ExpressionTree`.